vllm/v1/core/kv_cache_utils.py

```diff
@@ -158,8 +163,73 @@ def get_all_free_blocks(self) -> List[KVCacheBlock]:
         return ret
 
 
-def hash_block_tokens(parent_block_hash: Optional[int],
-                      curr_block_token_ids: Tuple[int]) -> BlockHashType:
+def generate_block_hash_extra_keys(
+        request: Request, start_token_idx: int, end_token_idx: int,
+        start_mm_idx: int) -> Tuple[Optional[Tuple[Any, ...]], int]:
+    """Generate extra keys for the block hash. The extra keys can come from
+    the multi-modal inputs and request specific metadata (e.g., LoRA ID).
+    For multi-modal inputs, the extra keys are (mm_hash, start_token_idx) that
```

Comment:
I think it's worth calling out `start_token_idx` here is the token index in the block rather than the request. (seems to be the case for current impl IIUC?)

Reply:
Good catch. This is actually a typo. The extra key is not `start_token_idx `. It's the starting offset of this image in this block.